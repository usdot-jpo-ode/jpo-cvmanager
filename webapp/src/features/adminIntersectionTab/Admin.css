#admin {
  width: 100%;
  height: calc(100vh - 135px);
}

.adminHeader {
  padding-top: 20px;
  padding-left: 25px;
  padding-bottom: 20px;
  text-align: left;
  color: white;
  font-family: sans-serif;
}

.errorMsg {
  max-width: 350px;
  color: #f83a25;
  padding: 2px 0;
  margin-top: 2px;
  font-size: 14px;
  font-weight: 300;
}

.react-tabs {
  display: flex;
  width: 100%;
  height: calc(100% - 77px);
  color: white;
  background: #1c1d1f;
}

.react-tabs__tab-list {
  display: flex;
  flex-direction: column;
  width: 170px;
  margin: 0;
  padding: 0;
  color: white;
  background: #333;
}

.react-tabs__tab {
  list-style: none;
  padding: 30px;
  cursor: pointer;
  font-weight: bold;
  font-size: large;
  color: #bbb;
  font-family: sans-serif;
}

.react-tabs__tab--selected {
  background: #0e2052;
  border-color: #1c1d1f;
  border-left: 4px solid #d16d15;
  color: white;
}

.react-tabs__tab-panel {
  display: none;
  width: 100%;
}

.react-tabs__tab-panel--selected {
  display: block;
}

.react-tabs__tab {
  padding-left: 25px;
}

.react-tabs__tab--selected {
  padding-left: 21px;
}

.panel-header {
  margin-bottom: 20px;
  padding: 5px;
  font-family: sans-serif;
  font-size: 25px;
  color: #d16d15;
  background-color: rgb(28, 29, 31);
  height: fit-content;
}

.panel-content {
  text-align: left;
  margin-top: 20px;
  margin-left: 30px;
  font-family: sans-serif;
}

.form-control {
  display: flex;
  flex-direction: column;
  margin-bottom: 15px;
  width: 400px;
  height: 30px;
  font-size: 15px;
}

.form-check {
  margin-bottom: 15px;
}

.form-select {
  margin-left: 10px;
  width: 150px;
}

.admin-button {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: 550;
  background-color: #b55e12;
  border: none;
  color: white;
  padding: 8px 10px;
  text-align: center;
  font-size: 12px;
  cursor: pointer;
  border-radius: 3px;
  max-width: 400px;
}

.admin_table_button {
  font-family: sans-serif;
  font-size: 15px;
  margin-top: -4px;
  margin-right: 10px;
  padding: 6px 6px;
  background-color: #b55e12;
  color: white;
  border: none;
  border-radius: 10px;
  cursor: pointer;
  text-align: center;
}

.org-form-test {
  margin-bottom: 10px;
}
.org-form-dropdown {
  width: 400px;
  max-width: 400px;
  margin-top: 5px;
  margin-bottom: 15px;
}

.form-multiselect,
.form-dropdown {
  width: 400px;
  max-width: 400px;
  margin-top: 3px;
  margin-bottom: 15px;
}

.org-multiselect {
  min-width: 300px;
  max-width: 95%;
  margin-right: 10px;
}

.spacer {
  margin-top: 0px;
  margin-bottom: 10px;
}

.spacer-large-intersection {
  margin-top: 0px;
  margin-bottom: 30px;
  display: flex;
}

.spacer-large-user {
  margin-top: 0px;
  margin-bottom: 30px;
  display: flex;
  align-items: center;
}

.column {
  flex: 50%;
}

.expand {
  color: #e98125;
}

.success-msg {
  color: rgb(0, 255, 0);
}

.error-msg {
  color: #f83a25;
  margin-bottom: 10px;
  background-color: #1c1d1f;
}

.accordion {
  margin-bottom: 10px;
}

.accordion-header {
  font-weight: bold;
}

.scroll-div {
  height: calc(100vh - 275px);
  overflow-y: auto;
  margin-right: 10px;
}

.scroll-div-tab {
  height: calc(100vh - 340px);
  overflow-y: auto;
}

.scroll-div-org-tab {
  height: calc(100vh - 390px);
  overflow-y: auto;
}

.admin_button {
  font-family: sans-serif;
  font-size: 15px;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-right: 10px;
  padding: 5px;
}

.plus_button {
  font-family: sans-serif;
  font-size: 15px;
  margin-top: -4px;
  margin-right: 10px;
  float: right;
  padding: 6px 6px;
  background-color: #b55e12;
  color: white;
  border: none;
  border-radius: 10px;
  cursor: pointer;
  text-align: center;
  max-height: 35px;
}

.delete_button {
  font-family: sans-serif;
  font-size: 15px;
  margin-top: 5px;
  margin-left: 10px;
  float: left;
  padding: 6px 6px;
  background-color: #b55e12;
  color: white;
  border: none;
  border-radius: 10px;
  cursor: pointer;
  text-align: center;
  max-height: 35px;
}

.table_div {
  width: calc(100% - 10px);
  overflow-wrap: normal;
  overflow: hidden;
}

@font-face {
  font-family: 'Material Icons';
  font-style: normal;
  font-weight: 400;
  src: url(https://fonts.gstatic.com/s/materialicons/v139/flUhRq6tzZclQEJ-Vdg-IuiaDsNc.woff2) format('woff2');
}

.material-icons {
  font-family: 'Material Icons';
  font-weight: normal;
  font-style: normal;
  font-size: 24px;
  line-height: 1;
  letter-spacing: normal;
  text-transform: none;
  display: inline-block;
  white-space: nowrap;
  word-wrap: normal;
  direction: ltr;
  -webkit-font-smoothing: antialiased;
}

.form-check-label {
  color: 'white';
}
