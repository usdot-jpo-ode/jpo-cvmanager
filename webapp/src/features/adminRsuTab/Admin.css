@import 'https://use.typekit.net/ioz0eth.css';

#admin {
  width: 100%;
  height: 100%;
}

.adminHeader {
  padding-top: 20px;
  padding-left: 25px;
  padding-bottom: 20px;
  text-align: left;
  font-family: 'museo-slab', sans-serif;
}

.errorMsg {
  max-width: 350px;
  color: #f83a25;
  padding: 2px 0;
  margin-top: 2px;
  font-size: 14px;
  font-weight: 300;
}

.panel-header {
  margin-bottom: 20px;
  padding: 5px;
  font-family: 'museo-slab', sans-serif;
  font-size: 25px;
  height: fit-content;
}

.panel-content {
  text-align: left;
  margin-top: 20px;
  margin-left: 30px;
  font-family: 'museo-slab', sans-serif;
}

.form-control {
  display: flex;
  flex-direction: column;
  margin-bottom: 15px;
  width: 400px;
  height: 30px;
  font-size: 15px;
}

.form-check {
  margin-bottom: 15px;
}

.form-select {
  margin-left: 10px;
  width: 150px;
}

.form-multiselect,
.form-dropdown {
  width: 400px;
  max-width: 400px;
  margin-top: 3px;
}

.org-multiselect {
  min-width: 200px;
}

.spacer {
  margin-top: 0px;
  margin-bottom: 10px;
}

.spacer-large-rsu {
  margin-top: 0px;
  margin-bottom: 30px;
  display: flex;
}

.spacer-large-user {
  margin-top: 0px;
  margin-bottom: 30px;
  display: flex;
  align-items: center;
}

.column {
  flex: 50%;
}

.accordion {
  border-radius: 0px;
}

.scroll-div {
  height: calc(100vh - 275px);
  overflow-y: auto;
  margin-right: 10px;
}

.scroll-div-tab {
  height: calc(100vh - 340px);
  overflow-y: auto;
}

.admin_button {
  font-family: 'museo-slab', sans-serif;
  font-size: 15px;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-right: 10px;
  padding: 5px;
}

.table_div {
  width: calc(100% - 10px);
  overflow-wrap: normal;
  overflow: hidden;
}

@font-face {
  font-family: 'Material Icons';
  font-style: normal;
  font-weight: 400;
  src: url(https://fonts.gstatic.com/s/materialicons/v139/flUhRq6tzZclQEJ-Vdg-IuiaDsNc.woff2) format('woff2');
}

.material-icons {
  font-family: 'Material Icons';
  font-weight: normal;
  font-style: normal;
  font-size: 24px;
  line-height: 1;
  letter-spacing: normal;
  text-transform: none;
  display: inline-block;
  white-space: nowrap;
  word-wrap: normal;
  direction: ltr;
  -webkit-font-smoothing: antialiased;
}
