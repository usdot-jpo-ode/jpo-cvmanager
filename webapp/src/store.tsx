import { Action, ThunkAction, configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import intersectionReducer from './generalSlices/intersectionSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddIntersectionReducer from './features/adminAddIntersection/adminAddIntersectionSlice'
import adminAddUserReducer from './features/adminAddUser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditIntersectionReducer from './features/adminEditIntersection/adminEditIntersectionSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminOrganizationTabIntersectionReducer from './features/adminOrganizationTabIntersection/adminOrganizationTabIntersectionSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminIntersectionTabReducer from './features/adminIntersectionTab/adminIntersectionTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import adminNotificationTabReducer from './features/adminNotificationTab/adminNotificationTabSlice'
import adminAddNotificationReducer from './features/adminAddNotification/adminAddNotificationSlice'
import adminEditNotificationReducer from './features/adminEditNotification/adminEditNotificationSlice'
import menuReducer from './features/menu/menuSlice'
import asn1DecoderSlice from './features/intersections/decoder/asn1-decoder-slice'
import intersectionMapReducer from './features/intersections/map/map-slice'
import intersectionMapLayerStyleReducer from './features/intersections/map/map-layer-style-slice'
import dataSelectorReducer from './features/intersections/data-selector/dataSelectorSlice'
import mapSliceReducer from './pages/mapSlice'

export const setupStore = (preloadedState: any) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      intersection: intersectionReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddIntersection: adminAddIntersectionReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditIntersection: adminEditIntersectionReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminOrganizationTabIntersection: adminOrganizationTabIntersectionReducer,
      adminRsuTab: adminRsuTabReducer,
      adminIntersectionTab: adminIntersectionTabReducer,
      adminUserTab: adminUserTabReducer,
      adminNotificationTab: adminNotificationTabReducer,
      adminAddNotification: adminAddNotificationReducer,
      adminEditNotification: adminEditNotificationReducer,
      menu: menuReducer,
      intersectionMap: intersectionMapReducer,
      intersectionMapLayerStyle: intersectionMapLayerStyleReducer,
      dataSelector: dataSelectorReducer,
      map: mapSliceReducer,
      asn1Decoder: asn1DecoderSlice,
    },
    preloadedState,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        thunk: true,
        serializableCheck: false,
        immutableCheck: false,
      }),
    devTools: true,
  })
}

type AppStore = ReturnType<typeof setupStore>
export type AppState = ReturnType<AppStore['getState']>
type AppThunk<ReturnType = void> = ThunkAction<ReturnType, AppState, unknown, Action>

export type RootState = ReturnType<ReturnType<typeof setupStore>['getState']>
