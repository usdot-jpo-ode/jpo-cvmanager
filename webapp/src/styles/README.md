# CVManager Theming

The CVManager natively supports 2 different themes: dark and light. Examples of each theme are shown below:

| Theme      | Map View                                                                       | Admin View                                                                         |
| ---------- | ------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------- |
| Main Dark  | <img src="./screenshots/main_dark_map.png" alt="Main Dark Map" width="300"/>   | <img src="./screenshots/main_dark_admin.png" alt="Main Dark Admin" width="300"/>   |
| Main Light | <img src="./screenshots/main_light_map.png" alt="Main Light Map" width="300"/> | <img src="./screenshots/main_light_admin.png" alt="Main Light Admin" width="300"/> |

## Light vs Dark

The CVManager is configured to automatically change theme and icons based on the browser being in light vs dark mode. Each of the theme components, the base MUI theme and the icons, have different versions for light and dark. If you don't want the theme to change with the browser, simply set both to the same theme.

## Themes

As noted above, the two built-in themes are light and dark. Each of these themes has an existing MUI Theme object in [index.ts](./index.ts). These theme objects look as follows:

```ts
const themeMainDark = createTheme({
  //MUI createTheme call, initialize all un-specified values to theme defaults: https://mui.com/material-ui/customization/default-theme/
  components: {
    MuiTableHead: {
      // Custom table headers
      styleOverrides: {
        root: {
          backgroundColor: '#252525', // Custom table headers background color
          borderBottom: 'none',
          '& .MuiTableCell-root': {
            borderBottom: 'none',
            fontSize: '12px',
            fontWeight: 600,
            lineHeight: 1,
            letterSpacing: 0.5,
            textTransform: 'uppercase',
          },
          '& .MuiTableCell-paddingCheckbox': {
            paddingTop: 4,
            paddingBottom: 4,
          },
        },
      },
    },
  },
  palette: {
    mode: 'dark', // Should match general color scheme, slightly changes rendering
    primary: {
      main: '#b55e12', // Primary color, used in icon backgrounds, selected tabs, buttons, checkboxes, ...
      light: '#cecece', // Used as LightButton background (contact support and logout)
      dark: '#e37120', // Highlighted/hovered button backgrounds, used in smaller buttons, admin, toggles. Provides better text contrast than primary.main
      contrastText: '#fff', // Text color with contrast to primary.main and primary.dark colors
    },
    secondary: {
      main: '#0e2052', // Secondary color, used in snmpwalk backgrounds
      light: '#26329f', // Used in snmpwalk hovered backgrounds
      dark: '#0a0f3f', // Used as accent color for selected tab item, snmpwalk  backgrounds, and LightButton text color (contact support and logout)
      contrastText: '#fff',
    },
    error: {
      main: '#FD7C7C', // Text color in RSU firmware and RSU status tables, as well as ErrorMessageText. Background is background.paper
    },
    success: {
      main: '#90EE90', // Text color in RSU firmware and RSU status tables, as well as SuccessMessageText. Background is background.paper
    },
    text: {
      primary: '#ffffff', // Primary text color, used in most text-based components
      secondary: '#eb8841', // Used for Intersection page sub-titles
      disabled: '#acacac', // Used for disabled button text
    },
    custom: {
      mapLegendBackground: '#0e2052', // Used for horizontal tab backgrounds and map control/legend backgrounds
      tableHeaderBackground: '#252525', // Used for Admin table header background color, usually same as components.styleOverrides.root.backgroundColor
      tableErrorBackground: '#4d2e2e', // Background color of admin table rows when there is an error
      mapStyleFilePath: 'mapbox-styles/cdot-dark.json', // path to local mapbox map style.json
      mapStyleHasTraffic: false, // Whether the style contains mapbox traffic and incident layers
    },
    divider: '#111', // Used in dividers (<Divider /> components)
    background: {
      paper: '#333', // Used in table body backgrounds, in-page header backgrounds, and vertical tab background. Also used in Help page background
      default: '#1c1d1f', // Used as default in page backgrounds
    },
  },
})
```

## Icons

The main icon/logo in the CVManager is used in 2 places: the after-login loading/error page, and the main dashboard at the top-left.

The CVManager allows customization of icons when building the application through docker. This is done by hosting the icons as volumes in the running webapp image, to the path container /usr/share/nginx/html/icons/logo_light.png and /usr/share/nginx/html/icons/logo_dark.png (path in repo is webapp/public/icons/logo_light and webapp/public/icons/logo_dark). This is already done in [docker-compose.yml](../../../docker-compose.yml#L79), and is set through the env vars WEBAPP_LOGO_PNG_ROOT_FILE_PATH_LIGHT and WEBAPP_LOGO_PNG_ROOT_FILE_PATH_DARK. These should each be set to the path of a .png image, which can be inside or outside of this repository.

Icons will be locked to 90px high when displayed.
