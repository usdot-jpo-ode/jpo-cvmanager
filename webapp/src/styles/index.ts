import { alpha, createTheme, Theme } from '@mui/material'
import './fonts/museo-slab.css'
import '../App.css'

declare module '@mui/material/styles' {
  interface PaletteOptions {
    custom: {
      mapLegendBackground: string
      tableHeaderBackground: string
      tableErrorBackground: string
      mapStyleFilePath: string
      mapStyleHasTraffic: boolean
      mapMenuItemBackgroundSelected: string
      mapMenuItemBorderSelected: string
      mapMenuItemHoverSelected: string
      mapMenuItemHoverUnselected: string
      mapMenuBackground: string
      intersectionMapAccordionExpanded: string
      intersectionMapButtonHover: string
      rowActionIcon: string
    }
  }
  interface Palette {
    custom: {
      mapLegendBackground: string
      tableHeaderBackground: string
      tableErrorBackground: string
      mapStyleFilePath: string
      mapStyleHasTraffic: boolean
      mapMenuItemBackgroundSelected: string
      mapMenuItemBorderSelected: string
      mapMenuItemHoverSelected: string
      mapMenuItemHoverUnselected: string
      mapMenuBackground: string
      intersectionMapAccordionExpanded: string
      intersectionMapButtonHover: string
      rowActionIcon: string
    }
  }
}

export const headerTabHeight = 119

export const testTheme = createTheme({
  palette: {
    custom: {
      mapLegendBackground: '#000000',
      tableHeaderBackground: '#000000',
      tableErrorBackground: '#000000',
      mapStyleFilePath: 'mapbox-styles/cdot-dark.json',
      mapStyleHasTraffic: true,
      mapMenuItemBackgroundSelected: '#000000',
      mapMenuItemBorderSelected: '#000000',
      mapMenuItemHoverSelected: '#000000',
      mapMenuItemHoverUnselected: '#000000',
      mapMenuBackground: '#000000',
      intersectionMapAccordionExpanded: '#000000',
      intersectionMapButtonHover: '#000000',
      rowActionIcon: '#000000',
    },
  },
})

// Please note that the light theme is currently not being maintained.
// Light Theme - https://www.realtimecolors.com/?colors=0a1424-e7eef8-213e73-7978d9-4431af&fonts=Inter-Inter
// --text: #0a1424;
// --background: #e7eef8;
// --primary: #213e73;
// --secondary: #7978d9;
// --accent: #4431af;
const themeMainLight = createTheme({
  cssVariables: true,
  typography: {
    fontFamily: '"Trebuchet MS", Arial, Helvetica, sans-serif',
    h1: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
    h2: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
    h3: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
    h4: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
    h5: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
    h6: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
  },
  components: {
    MuiTableHead: {
      styleOverrides: {
        root: {
          backgroundColor: '#F3F4F6',
          '.MuiTableCell-root': {
            color: '#374151',
          },
          borderBottom: 'none',
          '& .MuiTableCell-root': {
            borderBottom: 'none',
            fontSize: '12px',
            fontWeight: 600,
            lineHeight: 1,
            letterSpacing: 0.5,
            textTransform: 'uppercase',
          },
          '& .MuiTableCell-paddingCheckbox': {
            paddingTop: 4,
            paddingBottom: 4,
          },
        },
      },
    },
    MuiPaper: {
      styleOverrides: {
        root: {
          backgroundImage: 'unset',
        },
      },
    },
    MuiDivider: {
      styleOverrides: {
        root: {
          borderColor: '#061731',
        },
      },
    },
  },
  palette: {
    mode: 'light',
    primary: {
      main: '#6f69e0',
      light: '#2e3574',
      dark: '#a7a3e7',
      contrastText: '#0b041b',
    },
    secondary: {
      main: '#53aaf1',
      light: '#114875',
      dark: '#92bcde',
      contrastText: '#121212',
    },
    error: {
      main: '#713737',
    },
    success: {
      main: '#315131',
    },
    text: {
      primary: '#121828',
      secondary: '#061731',
      disabled: 'rgba(55, 65, 81, 0.48)',
    },
    divider: '#8f8f8f',
    background: {
      default: '#ececec',
      paper: '#FFFFFF',
    },
    custom: {
      mapLegendBackground: '#c8cfda',
      tableHeaderBackground: '#F3F4F6',
      tableErrorBackground: '#fdc7c7',
      mapStyleFilePath: 'mapbox-styles/main-light.json',
      mapStyleHasTraffic: true,
      mapMenuBackground: '#e0e0e0',
      mapMenuItemBackgroundSelected: '#c4c2c2',
      mapMenuItemBorderSelected: '1px solid black',
      mapMenuItemHoverSelected: '#a19f9f',
      mapMenuItemHoverUnselected: '#ffffff',
      intersectionMapAccordionExpanded: '#d7d7d7',
      intersectionMapButtonHover: '#d7d7d7',
      rowActionIcon: '#9DBDD3',
    },
  },
})

// --text: #dbe5f5;
// --background: #1b1d1f;
// --primary: #4383ad;
// --secondary: #dbe5f5;
// --accent: #614fcd;
const themeMainDark = createTheme({
  cssVariables: true,
  typography: {
    fontFamily: '"Trebuchet MS", Arial, Helvetica, sans-serif',
    h1: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
    h2: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
    h3: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
    h4: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
    h5: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
    h6: {
      fontFamily: '"museo-slab" Arial Helvetica Sans-Serif',
    },
  },
  components: {
    MuiTableHead: {
      styleOverrides: {
        root: {
          backgroundColor: '#252525',
          borderBottom: 'none',
          '& .MuiTableCell-root': {
            borderBottom: 'none',
            fontSize: '12px',
            fontWeight: 600,
            lineHeight: 1,
            letterSpacing: 0.5,
            textTransform: 'uppercase',
          },
          '& .MuiTableCell-paddingCheckbox': {
            paddingTop: 4,
            paddingBottom: 4,
          },
        },
      },
    },
    MuiPaper: {
      styleOverrides: {
        root: {
          backgroundImage: 'unset',
        },
      },
    },
    MuiDivider: {
      styleOverrides: {
        root: {
          borderColor: '#c7c7c7',
        },
      },
    },
  },
  palette: {
    mode: 'dark',
    primary: {
      main: '#3B7BA5',
      light: '#51a2d6',
      dark: '#3B7BA54D',
      contrastText: '#fff',
    },
    secondary: {
      main: '#dbe5f5',
      light: '#535297',
      dark: '#161563',
      contrastText: '#FAFAFA',
    },
    error: {
      light: '#FF6D57',
      main: '#E94F3766',
      dark: '#6e312a',
    },
    success: {
      light: '#75BD27',
      main: '#A0D36466',
      dark: '#9e0e0e',
    },
    text: {
      primary: '#FFFFFF',
      secondary: '#c7c7c7',
      disabled: '#acacac',
    },
    info: {
      main: '#dbe5f5',
    },
    custom: {
      mapLegendBackground: '#1b1d1f',
      tableHeaderBackground: '#252525',
      tableErrorBackground: '#4d2e2e',
      mapStyleFilePath: 'mapbox-styles/main-dark.json',
      mapStyleHasTraffic: true,
      mapMenuBackground: '#3c3c3c',
      mapMenuItemBackgroundSelected: '#333333',
      mapMenuItemBorderSelected: '1px solid black',
      mapMenuItemHoverSelected: '#333333',
      mapMenuItemHoverUnselected: '#575757',
      intersectionMapAccordionExpanded: '#2E2F31',
      intersectionMapButtonHover: '#2D5F7F',
      rowActionIcon: '#9DBDD3',
    },
    background: {
      paper: '#1b1d1f',
      default: '#333333',
    },
  },
})

// All available themes
export const THEMES = {
  light: themeMainLight,
  dark: themeMainDark,
}

export const getCurrentTheme = (isDarkTheme: boolean, lightThemeName: string, darkThemeName: string): Theme => {
  // Warnings to user if theme names are not known
  if (darkThemeName && !THEMES[darkThemeName]) {
    console.warn(`Unknown dark theme name: ${darkThemeName}`)
  }
  if (lightThemeName && !THEMES[lightThemeName]) {
    console.warn(`Unknown light theme name: ${lightThemeName}`)
  }
  if (isDarkTheme) {
    return THEMES[darkThemeName] ?? THEMES.dark
  } else {
    return THEMES[lightThemeName] ?? THEMES.light
  }
}
