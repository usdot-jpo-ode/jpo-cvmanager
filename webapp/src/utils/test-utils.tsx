/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container: HTMLElement) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val: string) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

import React from 'react'
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
import dayjs from 'dayjs'
import utc from 'dayjs/plugin/utc'
import timezone from 'dayjs/plugin/timezone'

// Extend dayjs with timezone and UTC plugins
dayjs.extend(utc)
dayjs.extend(timezone)

// Mock component to set timezone
const MockLocalizationProvider = ({ children }) => {
  dayjs.tz.setDefault('America/Denver')
  return <LocalizationProvider dateAdapter={AdapterDayjs}>{children}</LocalizationProvider>
}

export { replaceChaoticIds, MockLocalizationProvider }
