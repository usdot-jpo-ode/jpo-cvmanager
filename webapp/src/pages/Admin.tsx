import React, { useEffect } from 'react'
import { useDispatch } from 'react-redux'
import { updateTableData as updateRsuTableData } from '../features/adminRsuTab/adminRsuTabSlice'
import { updateTableData as updateIntersectionTableData } from '../features/adminIntersectionTab/adminIntersectionTabSlice'
import { getAvailableUsers } from '../features/adminUserTab/adminUserTabSlice'
import '../features/adminRsuTab/Admin.css'
import { AnyAction, ThunkDispatch } from '@reduxjs/toolkit'
import { RootState } from '../store'
import AdminOrganizationTab from '../features/adminOrganizationTab/AdminOrganizationTab'
import AdminRsuTab from '../features/adminRsuTab/AdminRsuTab'
import AdminUserTab from '../features/adminUserTab/AdminUserTab'
import { NotFound } from './404'
import { SecureStorageManager } from '../managers'
import { getUserNotifications } from '../features/adminNotificationTab/adminNotificationTabSlice'
import VerticalTabs from '../components/VerticalTabs'
import AdminIntersectionTab from '../features/adminIntersectionTab/AdminIntersectionTab'

function Admin() {
  const dispatch: ThunkDispatch<RootState, void, AnyAction> = useDispatch()

  useEffect(() => {
    dispatch(updateRsuTableData())
    dispatch(updateIntersectionTableData())
    dispatch(getAvailableUsers())
    dispatch(getUserNotifications())
  }, [dispatch])

  return (
    <>
      {SecureStorageManager.getUserRole() !== 'admin' ? (
        <div id="admin">
          <NotFound description="You do not have permission to view this page. Please return to main dashboard: " />
        </div>
      ) : (
        <div id="admin">
          <h2 className="adminHeader">CV Manager Admin Interface</h2>
          <VerticalTabs
            notFoundRoute={
              <NotFound
                redirectRoute="/dashboard/admin"
                redirectRouteName="Admin Page"
                description="This page does not exist. Please return to the main admin page."
              />
            }
            defaultTabIndex={0}
            tabs={[
              {
                path: 'rsus',
                title: 'RSUs',
                child: <AdminRsuTab />,
              },
              {
                path: 'intersections',
                title: 'Intersections',
                child: <AdminIntersectionTab />,
              },
              {
                path: 'users',
                title: 'Users',
                child: <AdminUserTab />,
              },
              {
                path: 'organizations',
                title: 'Organizations',
                child: <AdminOrganizationTab />,
              },
            ]}
          />
        </div>
      )}
    </>
  )
}

export default Admin
