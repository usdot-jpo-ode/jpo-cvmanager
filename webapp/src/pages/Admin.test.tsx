import React from 'react'
import { render } from '@testing-library/react'
import Admin from './Admin'
import { Provider } from 'react-redux'
import { ThemeProvider } from '@mui/material'
import { testTheme } from '../styles'
import { setupStore } from '../store'
import { replaceChaoticIds } from '../utils/test-utils'
import { BrowserRouter } from 'react-router-dom'

it('should take a snapshot', () => {
  const { container } = render(
    <ThemeProvider theme={testTheme}>
      <ThemeProvider theme={testTheme}>
        <Provider store={setupStore({})}>
          <BrowserRouter>
            <Admin />
          </BrowserRouter>
        </Provider>
      </ThemeProvider>
    </ThemeProvider>
  )
  replaceChaoticIds(container)

  expect(container).toMatchSnapshot()
})
