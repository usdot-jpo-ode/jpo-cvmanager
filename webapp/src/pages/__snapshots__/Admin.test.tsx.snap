// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    id="admin"
  >
    <div
      style="display: flex; justify-content: center; flex-direction: column; align-items: center; width: 100%; height: calc(100vh - 135px); background-color: rgb(21, 49, 126);"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-mocked-MuiTypography-root"
      >
        404 - Page Not Found
      </h1>
      <br />
      <h3
        class="MuiTypography-root MuiTypography-h3 css-mocked-MuiTypography-root"
      >
        You do not have permission to view this page. Please return to main dashboard: 
      </h3>
      <br />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-mocked-MuiButtonBase-root-MuiButton-root"
        style="font-size: 20px;"
        tabindex="0"
        type="button"
      >
        Main Dashboard
        <span
          class="MuiTouchRipple-root css-mocked-MuiTouchRipple-root"
        />
      </button>
      <h1
        style="color: white;"
      />
    </div>
  </div>
</div>
`;
